testing/jormungandr-integration-tests/src/common/jormungandr/process.rs

```diff
@@ -239,21 +240,55 @@ impl JormungandrProcess {
 
 impl Drop for JormungandrProcess {
     fn drop(&mut self) {
-        let errors = self
-            .logger
-            .get_lines_with_level(LogLevel::ERROR)
-            .collect::<Vec<_>>();
-        if !errors.is_empty() {
-            println!("Error lines:");
-            for line in errors {
-                println!("{}", line);
-            }
-        }
+        use rand::Rng;
         // There's no kill like overkill
         let _ = self.child.kill();
-
         // FIXME: These should be better done in a test harness
         self.child.wait().unwrap();
+        if panicking() {
+            let dir_name = rand::thread_rng()
+                .sample_iter(&rand::distributions::Alphanumeric)
+                .take(8)
+                .map(char::from)
+                .collect::<String>();
+
+            let logs_dir = std::env::temp_dir()
+                .join("jormungandr-logs")
+                .join(format!("jormungandr-{}", dir_name));
+
+            println!(
+                "persisting node temp_dir after panic: {}",
+                logs_dir.display()
+            );
+
+            // This code will not panic, as it will pollute the backtrace
+            // by doing so when the process is already panicking
+            std::fs::create_dir_all(&logs_dir)
+                .unwrap_or_else(|e| eprintln!("Could not create logs dir: {}", e));
+
+            if let Some(dir) = self.temp_dir.take() {
```

Comment:
Should be doable with `fs_extra`. Already imported in the project as well. Although I am not sure if it is actually used and is not a leftover dependency from long ago.

Reply:
(by this I mean that `rg fs_extra` gives me only `Cargo.toml` files)